hudi-client/src/main/java/org/apache/hudi/client/AutoCleanerService.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client;
+
+import org.apache.hudi.async.AbstractAsyncService;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.exception.HoodieException;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Auto Clean service running concurrently with write operation.
+ */
+class AutoCleanerService extends AbstractAsyncService {
+
+  private static final Logger LOG = LogManager.getLogger(AutoCleanerService.class);
+
+  private final HoodieWriteClient writeClient;
+  private final String cleanInstant;
+  private final transient ExecutorService executor = Executors.newFixedThreadPool(1);
+
+  protected AutoCleanerService(HoodieWriteClient writeClient, String cleanInstant) {
+    this.writeClient = writeClient;
+    this.cleanInstant = cleanInstant;
+  }
+
+  @Override
+  protected Pair<CompletableFuture, ExecutorService> startService() {
```

Comment:
This class feels more like a `AsyncTask` , rather than a service.. i.e something that is long running and accepts tasks.. can we file a follow on to clean this code uop?